### PR TITLE
integration-tests: Use the generic framework for development LP tests

### PR DESCRIPTION
**Pull request by cdamian**
# Description

Follow-up for the `.43` update work where only some LP integration tests were added to the generic framework.

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by lemunozm — COMMENTED** [review]
Thanks so much for all this work porting this 💯

File: runtime/integration-tests/src/generic/config.rs [hunk 1]
```diff
@@ -123,35 +151,48 @@ pub trait Runtime:
...
+		+ From<pallet_democracy::Event<Self>>;
 
 	type RuntimeOriginExt: Into<Result<RawOrigin<Self::AccountId>, <Self as frame_system::Config>::RuntimeOrigin>>
 		+ From<RawOrigin<Self::AccountId>>
 		+ Clone
 		+ OriginTrait<Call = <Self as frame_system::Config>::RuntimeCall>
 		+ From<pallet_ethereum::RawOrigin>
-		+ Into<Result<pallet_ethereum::Origin, <Self as frame_system::Config>::RuntimeOrigin>>;
+		+ Into<Result<pallet_ethereum::Origin, <Self as frame_system::Config>::RuntimeOrigin>>
+		+ From<pallet_liquidity_pools_gateway::GatewayOrigin>;
```
Comment (lemunozm) [inline runtime/integration-tests/src/generic/config.rs:195]: Suppose that it took a while to figure out this. 💯

File: runtime/integration-tests/src/generic/cases/liquidity_pools.rs [hunk 2]
```diff
@@ -26,126 +75,7098 @@ use xcm::{
...
+		crate::test_for_runtimes!([development], add_tranche);
+		crate::test_for_runtimes!([development], update_member);
+		crate::test_for_runtimes!([development], update_token_price);
+		crate::test_for_runtimes!([development], add_currency);
+		crate::test_for_runtimes!([development], add_currency_should_fail);
+		crate::test_for_runtimes!([development], allow_investment_currency);
+		crate::test_for_runtimes!([development], allow_pool_should_fail);
+		crate::test_for_runtimes!([development], schedule_upgrade);
+		crate::test_for_runtimes!([development], cancel_upgrade);
+		crate::test_for_runtimes!([development], update_tranche_token_metadata);
```
Comment (lemunozm) [inline runtime/integration-tests/src/generic/cases/liquidity_pools.rs:1777]: Out of curiosity, what's the current problem with the rest of the runtimes?

**Review by lemunozm — APPROVED** [review]
Amazing PR and work @cdamian! 🚀 

Really love how things are now organized in integration-tests, thanks a lot for all your effort on this! 🥇

File: runtime/integration-tests/src/generic/config.rs (hunk 1 above)
Comment (cdamian) [reply runtime/integration-tests/src/generic/config.rs:195]: This, not that much, luckily the compiler was super helpful here. I had more fun on the events side ^^

File: runtime/integration-tests/src/generic/cases/liquidity_pools.rs (hunk 2 above)
Comment (cdamian) [reply runtime/integration-tests/src/generic/cases/liquidity_pools.rs:1777]: I think it's mostly related to the native currency that we use there, and probably some other things that @NunoAlexandre might know better. However, I also think that later on we can probably improve these tests a bit more and try to run them for all runtimes, I'm sure there's a way to achieve that.

**Pull request merged**